xml_analyzer/xmlreader.py

```diff
@@ -71,21 +70,31 @@ def resolve_dependencies(self, binding_connector_list: list) -> list:
 
         return dependencies
 
-    def find_constraint_spec(self, constraint_property: Et.Element) -> str:
+    def _find_constraint_spec(self, constraint_property: Et.Element) -> str:
         query = ".//ConstraintElement/ConstrainedElements//*[@Idref='{}']/../..//CompositeValueSpecification"\
             .format(constraint_property.get("Id"))
         constraint_spec = self.root.find(query).get('Value')
         return constraint_spec
 
-    def build_data_container(self, constraint_property: Et.Element) -> DataContainer:
+    def _find_constraint_property(self, val: str, method='by_id'):
+        element = None
+        if method == 'by_id':
+            query = ".//SysMLConstraintProperty[@Id='{}']".format(val)
+            element = self.root.find(query)
+
+        return element
+
+    def build_data_container(self, constraint_property_id: str) -> DataContainer:
```

Comment:
method is very long you can easily factor out the for loop

Reply:
Fixed with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/125a9de3f5317b3cc4f907790a9fabccc9134c0e